examples/demo/src/orders/OrderList.tsx

```diff
@@ -152,169 +154,127 @@ const TabbedDatagrid = () => {
-                                source="basket.length"
-                                label="resources.orders.fields.nb_items"
-                            />
-                            <NumberField
-                                source="total_ex_taxes"
-                                options={{
-                                    style: 'currency',
-                                    currency: 'USD',
-                                }}
-                            />
-                            <NumberField
-                                source="delivery_fees"
-                                options={{
-                                    style: 'currency',
-                                    currency: 'USD',
-                                }}
-                            />
-                            <NumberField
-                                source="taxes"
-                                options={{
-                                    style: 'currency',
-                                    currency: 'USD',
-                                }}
-                            />
-                            <NumberField
-                                source="total"
-                                options={{
-                                    style: 'currency',
-                                    currency: 'USD',
-                                }}
-                                sx={{ fontWeight: 'bold' }}
-                            />
-                        </DatagridConfigurable>
+                        <CancelledOrdersTable />
                     )}
                 </>
             )}
         </Fragment>
     );
 };
 
+const Column = DataTable.Col<Order>;
+const ColumnNumber = DataTable.NumberCol<Order>;
+
+const OrderedOrdersTable = React.memo(() => (
+    <DataTable
+        rowClick="edit"
+        hiddenColumns={['total_ex_taxes', 'delivery_fees', 'taxes']}
+        storeKey={storeKeyByStatus.ordered}
+    >
+        <Column source="date">
+            <DateField source="date" showTime />
+        </Column>
+        <Column source="reference" />
+        <Column source="customer_id" field={CustomerReferenceField} />
+        <Column label="resources.orders.fields.address">
+            <ReferenceField
+                source="customer_id"
+                reference="customers"
+                link={false}
+            >
+                <AddressField />
+            </ReferenceField>
+        </Column>
+        <ColumnNumber
+            source="basket.length"
+            label="resources.orders.fields.nb_items"
+        />
+        <ColumnNumber source="total_ex_taxes" options={currencyStyle} />
+        <ColumnNumber source="delivery_fees" options={currencyStyle} />
+        <ColumnNumber source="taxes" options={currencyStyle} />
+        <ColumnNumber
+            source="total"
+            options={currencyStyle}
+            sx={{ fontWeight: 'bold' }}
+        />
+    </DataTable>
```

Comment:
If I'm not mistaken, all tables have the same columns except `DeliveredOrdersTable` that adds the `returned` column. Can't we have a `CommonColumn` const that has a `Fragment` containing the common cols. That's a good test and showcase of what the new `DataTable` allows too!

Reply:
fixed